File: app/src/main/java/me/ycdev/android/devtools/PhoneApp.java
package me.ycdev.android.devtools;

import android.app.Application;
import android.os.Process;

import timber.log.Timber;

public class PhoneApp extends Application {
    private static final String TAG = "DevToolsApp";

    @Override
    public void onCreate() {
        super.onCreate();
        Timber.plant(new Timber.DebugTree());
        Timber.tag(TAG).i("app start...");
        checkAppReplacingState();
    }

    private void checkAppReplacingState() {
        if (getResources() == null) {
            Timber.tag(TAG).w("app is replacing...kill");
            Process.killProcess(Process.myPid());
        }
    }
}

File: app/src/main/java/me/ycdev/android/devtools/PhoneApp.kt
package me.ycdev.android.devtools

import android.app.Application
import android.os.Process
import timber.log.Timber
import timber.log.Timber.DebugTree

class PhoneApp : Application() {
    override fun onCreate() {
        super.onCreate()
        Timber.plant(DebugTree())
        Timber.tag(TAG).i("app start...")
        checkAppReplacingState()
    }

    private fun checkAppReplacingState() {
        if (resources == null) {
            Timber.tag(TAG).w("app is replacing...kill")
            Process.killProcess(Process.myPid())
        }
    }

    companion object {
        private const val TAG = "DevToolsApp"
    }
}
